Makefile

```diff
@@ -1,18 +1,56 @@
 GOTOOLS = \
-					github.com/mitchellh/gox \
-					github.com/Masterminds/glide \
-					gopkg.in/alecthomas/gometalinter.v2 \
-					github.com/gogo/protobuf/protoc-gen-gogo \
-					github.com/gogo/protobuf/gogoproto
-
+	github.com/mitchellh/gox \
+	github.com/Masterminds/glide \
+	gopkg.in/alecthomas/gometalinter.v2 \
+	github.com/gogo/protobuf/protoc-gen-gogo \
+	github.com/gogo/protobuf/gogoproto
+GOTOOLS_CHECK = gox glide gometalinter.v2 protoc protoc-gen-gogo
+PACKAGES=$(shell go list ./... | grep -v '/vendor/')
 INCLUDE = -I=. -I=${GOPATH}/src -I=${GOPATH}/src/github.com/gogo/protobuf/protobuf
 
-all: protoc install test metalinter
+all: check build test install metalinter
 
-PACKAGES=$(shell go list ./... | grep -v '/vendor/')
+check: check_tools get_vendor_deps
+
+
+########################################
+###  Build
 
-install_protoc:
-	# https://github.com/google/protobuf/releases
+protoc:
+	## If you get the following error,
+	## "error while loading shared libraries: libprotobuf.so.14: cannot open shared object file: No such file or directory"
+	## See https://stackoverflow.com/a/25518702
+	protoc $(INCLUDE) --gogo_out=plugins=grpc:. types/*.proto
+	@echo "--> adding nolint declarations to protobuf generated files"
+	@awk '/package types/ { print "//nolint: gas"; print; next }1' types/types.pb.go > types/types.pb.go.new
+	@mv types/types.pb.go.new types/types.pb.go
+
+build: protoc
+	@go build -i ./cmd/...
+
+dist:
+	@bash scripts/dist.sh
+	@bash scripts/publish.sh
+
+install:
+	@go install ./cmd/...
+
+
+########################################
+### Tools & dependencies
+
+check_tools:
+	@# https://stackoverflow.com/a/25668869
+	@echo "Found tools: $(foreach tool,$(GOTOOLS_CHECK),\
+        $(if $(shell which $(tool)),$(tool),$(error "No $(tool) in PATH")))"
+
+get_tools:
+	@echo "--> Installing tools"
+	go get -u -v $(GOTOOLS)
+	@gometalinter.v2 --install
+
+get_protoc:
+	@# https://github.com/google/protobuf/releases
```

Comment:
Why the commented directives?

Reply:
Otherwise it prints onto the console, as if you typed it into bash.